packages/jsii-kernel/lib/kernel.ts

```diff
@@ -521,10 +531,16 @@ export class Kernel {
         return `$jsii$super$${name}$`;
     }
 
-    private _applyPropertyOverride(obj: any, objref: api.ObjRef, override: api.Override) {
+    private _applyPropertyOverride(obj: any, objref: api.ObjRef, override: api.Override, propInfo?: spec.Property) {
         const self = this;
         const propertyName = override.property!;
 
+        // if this is a private property (i.e. doesn't have `propInfo` the object has a key)
+        if (!propInfo && propertyName in obj) {
+            this._debug(`Skipping override of private property ${propertyName}`);
```

Comment:
Shouldn't this `throw`? I can't help but think this is the symptom of a bug...

Reply:
Not necessarily, there could be case where the user would define a public method in a derived native class with the same name as a private method in a base javascript class. When the native jsii client instantiates the javascript object, it doesn't know that this method is private on the javascript side, do it will pass it as an override (you could argue that the jsii client should only pass overrides for known public methods, but currently this is not the behavior, at least in java).

Reply:
That makes sense to me. I think the front-end code should be a little more educated about the type model it was generated for... So I reckon this should be a `WARN` and not a `DEBUG`.